dotCMS/src/main/java/com/dotmarketing/business/BaseWebAssetAPI.java

```diff
@@ -476,40 +477,46 @@ public int deleteOldVersions(final Date olderThan, final String type) throws Dot
 
         //No need to run all the SQL is we don't get any inodes from the condition.
         if(!resultInodes.isEmpty()){
+            String parameterPlaceholders;
+            List<String> inodesToDeleteList = new ArrayList<>();

```

Comment:
It seems you are getting twice memory, could set to null or Collections.EMPTY_LIST